agent/config/types.go

```diff
@@ -15,11 +15,20 @@ package config
 
 import (
 	"encoding/json"
+	"log"
+	"strconv"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/engine/dockerclient"
 )
 
+const (
+	_ DefaultBool = iota
```

Comment:
`DefaultBool` doesn't capture the abstraction intuitively imho.

Reply:
`DefaultBool` is a terrible and ambiguous name. I'm not sure we would want to name it `Tri*` that on the chance we decide to add fourth state `DefaultDisabled`. 

Is `Conditional` reasonable here?

Reply:
`Conditional` appears to be reasonable here 👍